spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -0,0 +1,233 @@
+				if (properties.isDeleteFilesOnExit()) {
+					workDir.toFile().deleteOnExit();
+				}
+
+				processes.add(new Instance(moduleDeploymentId, i, builder, workDir));
+				logger.info("deploying module {} instance {}\n   Logs will be in {}", module, i, workDir);
+			}
+			catch (IOException e) {
+				throw new RuntimeException("Exception trying to deploy " + request, e);
+			}
+
+		}
+
+
+		return moduleDeploymentId;
+	}
+
+	@Override
+	public void undeploy(ModuleDeploymentId id) {
+		List<Instance> processes = running.get(id);
+		if (processes != null) {
+			RestTemplate restTemplate = new RestTemplate();
+			for (Instance instance : processes) {
+				if (!isAlive(instance.process)) {
+					continue;
+				}
+				try {
+					restTemplate.postForObject(instance.moduleUrl + "/shutdown", null, String.class);
+					instance.process.waitFor();
+				}
+				catch (InterruptedException e) {
+					instance.process.destroy();
+				}
+			}
+			running.remove(id);
+		}
+	}
+
+	@Override
+	public ModuleStatus status(ModuleDeploymentId id) {
+		List<Instance> instances = running.get(id);
+		if (instances == null) {
+			return null;
+		}
+		ModuleStatus.Builder builder = ModuleStatus.of(id);
+		for (Instance instance : instances) {
+			builder.with(instance);
+		}
+		return builder.build();
+	}
+
+	@Override
+	public Map<ModuleDeploymentId, ModuleStatus> status() {
+		Map<ModuleDeploymentId, ModuleStatus> result = new HashMap<>();
+		for (Map.Entry<ModuleDeploymentId, List<Instance>> entry : running.entrySet()) {
+			ModuleStatus.Builder builder = ModuleStatus.of(entry.getKey());
+			for (Instance instance : entry.getValue()) {
+				builder.with(instance);
+			}
+			result.put(entry.getKey(), builder.build());
+		}
+		return result;
+	}
+
+	/**
+	 * Copies the module launcher jar somewhere at startup, so that it can be invoked later
+	 * via {@literal java -jar}.
+	 */
+	@PostConstruct
+	public void setup() throws IOException {
+		File launcherFile = Files.createTempFile("spring-cloud-stream-module-launcher", ".jar").toFile();
+		launcherFile.deleteOnExit();
+		FileCopyUtils.copy(new ClassPathResource("spring-cloud-stream-module-launcher.jar").getInputStream(), new FileOutputStream(launcherFile));
+		this.moduleLauncherPath = launcherFile.getAbsolutePath();
+	}
+
+	private static class Instance implements ModuleInstanceStatus {
+		private final ModuleDeploymentId moduleDeploymentId;
+
+		private final int instanceNumber;
+
+		private final Process process;
+
+		private final File workDir;
+
+		private final File stdout;
+
+		private final File stderr;
+
+		private final URL moduleUrl;
+
+		private Instance(ModuleDeploymentId moduleDeploymentId, int instanceNumber, ProcessBuilder builder, Path workDir) throws IOException {
+			this.moduleDeploymentId = moduleDeploymentId;
+			this.instanceNumber = instanceNumber;
+			builder.directory(workDir.toFile());
+			builder.redirectOutput(this.stdout = Files.createTempFile(workDir, "stdout_", ".log").toFile());

```

Comment:
Need to register the file to `deleteOnExit()` or else the directory will not be deleted either.